src/android/test.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * test module
+ *
+ * Created by Appcelerator, Inc
+ * Copyright (c) 2016 Appcelerator. All rights reserved.
+ */
+import { View } from './view';
+
+export default class Test {
+
+	constructor (app, module) {
+		this.app = app;
+		this.module = module;
+	}
+
+	startup () {
+		this.app.logger.info(this.module.id + ' startup');
+	}
+
+	shutdown () {
+		this.app.logger.info(this.module.id + ' shutdown');
+	}
+
+	static createView (opts = {}) {
```

Comment:
Yes, I kept the current factory approach that we're using in Ti modules today but I'm not sure it's necessary. in fact, if we take your approach, you wouldn't need this class at all until you wanted to hook into startup.